.github/workflows/unit-test-tools.yml

```diff
@@ -0,0 +1,106 @@
+name: unit-test-tools
+
+permissions: read-all
+
+on:
+  pull_request:
+    paths:
+      - 'tools/**'
+      - '.github/workflows/unit-test-tools.yml'
+
+jobs:
+  unit-test-diff-processor:
```

Comment:
the `unit-test` part of these job names is redundant because of how these are displayed in the UI:

<img width="421" alt="Screenshot 2024-05-16 at 9 52 36 AM" src="https://github.com/GoogleCloudPlatform/magic-modules/assets/299979/7864a035-c7cf-4c87-950b-90fa0c36c5c4">


```suggestion
  diff-processor:
```